src/kernel/vatManager.js

```diff
@@ -138,128 +98,53 @@ export default function makeVatManager(vatID, syscallManager, setup, helpers) {
   // just did syscall.send and referenced 'slot' (in an argument, or as the
   // target of a send), what are they talking about?
 
-  function mapOutbound(slot) {
-    // kdebug(`mapOutbound ${JSON.stringify(slot)}`);
-    if (slot.type === 'export') {
+  // vatSlot to kernelSlot
+  // mapOutbound
+  function mapVatSlotToKernelSlot(vatSlot) {
+    vatState.checkVatSlot(vatSlot);
+    const { type, id } = vatSlot;
+
+    if (type === 'export') {
       // one of our exports, so just make the vatID explicit
-      Nat(slot.id);
-      return { type: 'export', vatID, id: slot.id };
+      return { type: 'export', vatID, id };
     }
 
-    if (slot.type === 'import') {
+    if (type === 'import' || type === 'deviceImport') {
```

Comment:
all four `mapVatSlotToKernelSlot` cases should insist upon the target being present, so this can get merged into the next clause